src/function/table/arrow.cpp

```diff
@@ -42,8 +42,25 @@ static unique_ptr<ArrowType> GetArrowLogicalTypeNoDictionary(ArrowSchema &schema
 		return make_uniq<ArrowType>(LogicalType::DOUBLE);
 	} else if (format[0] == 'd') { //! this can be either decimal128 or decimal 256 (e.g., d:38,0)
 		std::string parameters = format.substr(format.find(':'));
```

Comment:
These format strings come to us from an external source, so I think it would make sense to be defensive here. We never know if some library could produce just `d` as a schema. Could we check that `format.find(':')` and `parameters.find(',')` return valid positions and throw if they do not?

Reply:
Perhaps using `StringUtil::Split` is easier to use in a defensive way since then you don't need to do all this arithmetic on string positions which is somewhat error prone

Reply:
Excellent idea